src/core/identifiers/identifiers.ts

```diff
@@ -75,8 +76,8 @@ export const convertBigNumberToDSNPUserId = (num: BigNumber): DSNPUserId => num.
  */
 export const convertDSNPUserIdOrURIToBigNumber = (userIdOrUri: DSNPUserId | DSNPUserURI): BigNumber => {
   // Support DSNP User URI
-  const hex = userIdOrUri.replace("dsnp://", "");
-  return BigNumber.from(hex);
+  const hex = userIdOrUri.replace(/dsnp:\/\/|0x/gi, "");
```

Comment:
why do we remove the `0x` only to add it back in the next line?

Reply:
also, why is this regex global? there should only ever be one DSNP prefix

Reply:
on and the `^` thing too

Reply:
Global drop 👍 
^ 👍 
Remove to add? Because the regex is using it to drop leading zeros and it is a more complex regex to only drop the leading zeros.